src/Microsoft.OData.Core/ODataMetadataJsonOutputContext.cs

```diff
@@ -175,22 +175,54 @@ internal static Utf8JsonWriter CreateJsonWriter(Stream stream, ODataMessageWrite
         /// </summary>
         /// <param name="disposing">If 'true' this method is called from user code; if 'false' it is called by the runtime.</param>
         protected override void Dispose(bool disposing)
+        {
+            if (disposing)
+            {
+                try
+                {
+                    if (this.jsonWriter != null)
+                    {
+                        if (this.asynchronousOutputStream != null)
+                        {
+                            DisposeOutputStreamAsync().Wait();

```

Comment:
It would be better to avoid the **sync over async** .Wait() call here - dispose the stream synchronously for the Dispose() implementation while the DisposeAsync() implementation can do it properly (asynchronously).

Reply:
Basically, I would try make sure of employing two best practices:
1. No use of **sync over async** (i.e., Task.Wait(), Task.Result, Task.GetAwaiter().GetResult()). In the Dispose() implementation just call synchronous Flush().
2. No use of synchronous Flush() in the async dispose implementation (DisposeAsyncCore). Always uses **await FlushAsync();**